api/src/pcapi/core/users/api.py

```diff
@@ -1264,28 +1264,31 @@ def is_login_device_a_trusted_device(
     return False
 
 
-def create_suspicious_login_email_token(login_info: users_models.LoginDeviceHistory | None, user_id: int) -> str:
-    current_datetime = datetime.datetime.utcnow()
-    expiration_date = current_datetime + datetime.timedelta(weeks=1)
-
+def create_suspicious_login_email_token(
+    login_info: users_models.LoginDeviceHistory | None, user_id: int
+) -> token_utils.Token:
     if login_info is None:
-        return users_utils.encode_jwt_payload(
-            token_payload={
-                "userId": user_id,
-                "dateCreated": current_datetime.strftime(date_utils.DATE_ISO_FORMAT),
-            },
-            expiration_date=expiration_date,
+        return token_utils.Token.create(
+            token_utils.TokenType.SUSPENSION_SUSPICIOUS_LOGIN,
+            users_constants.SUSPICIOUS_LOGIN_EMAIL_TOKEN_LIFE_TIME,
+            user_id,
+            {"dateCreated": datetime.datetime.utcnow().strftime(date_utils.DATE_ISO_FORMAT)},
         )
 
-    return users_utils.encode_jwt_payload(
-        token_payload={
-            "userId": user_id,
+    return token_utils.Token.create(
+        token_utils.TokenType.SUSPENSION_SUSPICIOUS_LOGIN,
+        max(
+            (login_info.dateCreated - datetime.datetime.utcnow())
+            + users_constants.SUSPICIOUS_LOGIN_EMAIL_TOKEN_LIFE_TIME,
+            datetime.timedelta(days=0),
+        ),
+        user_id,
+        {
             "dateCreated": login_info.dateCreated.strftime(date_utils.DATE_ISO_FORMAT),
             "location": login_info.location,
             "os": login_info.os,
             "source": login_info.source,
         },
-        expiration_date=expiration_date,
     )
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏J'ai du mal à comprendre ce morceau de code
Ça mérite un commentaire, ou une fonction bien nommée avec la logique dedans je trouve

Reply:
la date expiration du token et calculé à partir de la date du login sucpicieux et pas de la date de création du token. mais redis prend en entrée un ttl donc je calcul le ttl et je fait le max entre ttl 0 et 0 car redis refuse un ttl negatif.
j'aurais peux être pu ne pas générer de token et retourner une erreur.